cmd/link/link.go

```diff
@@ -0,0 +1,41 @@
+package link
+
+import (
+	"fmt"
+
+	"github.com/ncw/rclone/cmd"
+	"github.com/ncw/rclone/fs/operations"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	cmd.Root.AddCommand(commandDefintion)
+}
+
+var commandDefintion = &cobra.Command{
+	Use:   "link remote:path",
+	Short: `Generate public link to file/folder.`,
+	Long: `
+rclone link will create or retrieve a public link to the given file or folder.
+
+    rclone link remote:path/to/file
+    rclone link remote:path/to/folder/
+
+If successful, the last line of the output will contain the link. Exact
+capabilities depend on the remote, but the link will always be created with
+the least constraints – e.g. no expiry, no password protection, accessible
+without account.
+`,
+	Run: func(command *cobra.Command, args []string) {
+		cmd.CheckArgs(1, 1, command, args)
+		fsrc, srcFileName := cmd.NewFsSrcWithFile(args)
+		cmd.Run(false, false, command, func() error {
+			link, err := operations.PublicLink(fsrc, srcFileName)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("public link:\n%s\n", link)
```

Comment:
This would probably be most useful for scripting if it is just the link without the `public link` text.

Reply:
Just putting the link out there seemed wrong, and there always is `tail -n1`. On second thought the user pretty much expects a link, so the explanation is superfluous. I've removed it.